pkg/reconciler/reconciler.go

```diff
@@ -0,0 +1,15 @@
+package reconciler
+
+// ReconcileStageFunc is a function reconciling an entity and returning an error.
+type ReconcileStageFunc func() error
+
+// ExecuteStages executes each of the given ReconcileStageFuncs one after the
+// other. Execution is aborted on the first error, which will be returned.
+func ExecuteStages(funcs ...ReconcileStageFunc) error {
```

Comment:
How about we do `func ExecuteStages(instance servingv1alpha1.Install, funcs ...ReconcileStageFunc) error {` and pass the instance to each function. Kinda enforces the idea that config might ultimately be in the `Install`'s spec.